Added rediscluster backend support and custom amqp broker

sunny316 opened the pull request.

sickyoon commented:
Could you separate redis-go implementation and amqp broker implementation (this has been addressed already)

sunny316 commented:
Which has been addressed ? Former or the later. 
If later has been addressed, do you want pull request with backend implementation for redis cluster ? I am using github.com/go-redis/redis/v7 for connecting to redis cluster as redigo does not support redis cluster. 
Thanks

sunny316 commented:
@sickyoon  I just went through the recent commits. I see that you have made the variables Public. 
The method in amqp_broker.go `NewAMQPCeleryBrokerByConnAndChannel` still uses hardcoded defaults for Exchange name and queue name. Also, the Publish method call in the `SendCeleryMessage` hard codes the first parameter to empty string which should be the exchange name. 
Thus, I added a new NewCustomAMQPCeleryBroker for custom Exchange and queue name. Also, added queuename to Publish method call. You can refer to the changes in the PR. If changes look good, I can create a separate PR for just AMQP broker changes.

sunny316 commented:
Find New PR for #151 here #152 
Right now it just has backend for redis cluster

sickyoon commented:
> @sickyoon I just went through the recent commits. I see that you have made the variables Public.
> The method in amqp_broker.go `NewAMQPCeleryBrokerByConnAndChannel` still uses hardcoded defaults for Exchange name and queue name. Also, the Publish method call in the `SendCeleryMessage` hard codes the first parameter to empty string which should be the exchange name.
> Thus, I added a new NewCustomAMQPCeleryBroker for custom Exchange and queue name. Also, added queuename to Publish method call. You can refer to the changes in the PR. If changes look good, I can create a separate PR for just AMQP broker changes.

Yes you are right. But with public variables, you don't really have to use `NewAMQPCeleryBrokerByConnAndChannel`. I prefer it if you could avoid adding additional constructors. Good catch with `SendCeleryMessage`. It is indeed a problem. The queue name should be pulled from `broker.QueueName`. I would love to review if you could remove the Redis cluster part.

sunny316 commented:
Will do
